src/fides/api/alembic/migrations/versions/5f96f13be8ec_property_specific_messaging_migration_.py

```diff
@@ -0,0 +1,189 @@
+            "INSERT INTO plus_property (id, name, is_default, type)"
+            "VALUES (:id, :name, :is_default, :type)"
+        )
+        characters = string.ascii_uppercase + string.digits
+        # fixme- does this need the plu_ prefix?
+        new_property_id: str = "plu_FDS-" + "".join(random.choices(characters, k=6))
+        default_property_id = new_property_id
+
+        new_property: Dict[str, Any] = {
+            "id": new_property_id,
+            "name": "Default Property",
+            "is_default": True,
+            "type": "Website",
+        }
+        try:
+            bind.execute(insert_into_property_query, new_property)
+        except IntegrityError as exc:
+            raise Exception(
+                f"Fides attempted to create a new default property but got error: {exc}. "
+            )
+
+    # STEP 2: Ensure all default templates are saved to the DB
+    existing_templates: ResultProxy = bind.execute(
+        text("select id, type from messaging_template;")
+    )
+    templates_from_db = [template.type for template in existing_templates.all()]
+    for template_type, template in DEFAULT_MESSAGING_TEMPLATES.items():
+        # if a template is not already in the DB, save it with defaults
+        if template_type not in templates_from_db:
+            insert_into_messaging_template_query: TextClause = text(
+                "INSERT INTO messaging_template (id, type, content, is_enabled)"
+                "VALUES (:id, :type, :content, :is_enabled)"
+            )
+            new_messaging_template: Dict[str, Any] = {
+                "id": generate_record_id("mes"),
+                "type": template_type,
+                "content": template["content"],
+                "is_enabled": False,
+            }
+            try:
+                bind.execute(
+                    insert_into_messaging_template_query.bindparams(
+                        bindparam("content", type_=JSONB)
+                    ),
+                    new_messaging_template,
+                )
+            except IntegrityError as exc:
+                raise Exception(
+                    f"Fides attempted to create a new messaging_template but got error: {exc}. "
+                )
+
+    # STEP 3: Ensure all saved messaging templates are linked to the default property
+    updated_templates: ResultProxy = bind.execute(
+        text("select id, type from messaging_template;")
+    )
+    for template in updated_templates:
+        insert_into_messaging_template_to_property_query: TextClause = text(
+            "INSERT INTO messaging_template_to_property (id, messaging_template_id, property_id)"
+            "VALUES (:id, :messaging_template_id, :property_id)"
+        )
+        new_messaging_template_to_property: Dict[str, Any] = {
+            "id": generate_record_id("mes"),
+            "messaging_template_id": template["id"],
+            "property_id": default_property_id,
```

Comment:
Litte more context: A customer must explicitly turn on the property specific messaging as a feature, via a new ENV var (will be done in this ticket- https://ethyca.atlassian.net/browse/PROD-2049). 

If that ENV var is `False`, when we send out emails from Fides, we get the 1st template by type, basically ignoring the property id. 

Before this ENV is enabled, the customer is expected to have to manually configure and explicitly enable their property-specific messaging templates in Admin-UI. 

That said, here in the migration, we can probably get away with not assigning a property at all to the default templates to keep things simpler here. Let me confirm with @mfbrown

Reply:
Actually, just remembered, there's a reason for assigning default property to all templates- in the edge case where: 

1. We do not assign default property to templates
2. Customer adds a new template with same type as an existing template, say `subject identity verification`, with property A. So now we have 1 `subject identity verification` template with no properties, and another `subject identity verification` template with property A. Both templates are enabled.
3. Customer turns OFF the ENV variable, so now Fides is in non-property-specific messaging mode.
4. At runtime, Fides needs to send out an email, but there are 2 templates with type `subject identity verification`
5. Which one do we use?

 We need to guarantee that we have a reliable way to send emails for the non-property-specific use case. If we enforce the fact that there is always 1 default to fall back on, this becomes simpler imo. Thoughts?

Reply:
I still feel like the particular use case I brought up reveals inconsistent behavior, where if they had one property, their property became linked to the templates, but if they had more than one, it did not.

Current behavior?
Customer has 0 properties -> Fides generated default property created and linked to the templates
Customer has 1 property  -> Use-created property is labeled as the default, and linked to the templates
Customer has 2 properties -> Fides generated default property is created and linked to the templates.

In the case of two properties, if you always need at least one default property (and we can't decide which one they would want to be a default), I'd maybe create a default property and link it to the templates, but also link their existing properties to the templates as well?

Open for discussion of course

Reply:
For now let's not link all existing properties to templates since the user can always configure whichever properties they need through the Admin-UI. We are additionally discussing some functionality where the templates linked to the default property might act as a catch-all at run-time for email requests that come in from un-configured properties. See #7 here- https://ethyca.atlassian.net/wiki/spaces/EN/pages/3055681577/2024-05-09+Property+Specific+Messaging#Open-questions%3A

We can follow-up as needed.